apps/designer/app/designer/features/style-panel/shared/css-value-input/css-value-input.tsx

```diff
@@ -77,6 +77,13 @@ const useScrub = ({
   const type = valueRef.current.type;
   const unit = type === "unit" ? valueRef.current.unit : undefined;
 
+  // make sure caret doesn't jump around as you scrub
+  useLayoutEffect(() => {
+    if (isInputActive) {
+      inputRef.current?.select();
```

Comment:
This is going to select the text every time we scrub, the previous behavior was nicer since you could see the text without the selection, which makes the text less readable.

Was there no other way? At this point I am not sure which one is worse the jumping caret next to the number or the always selected text.

Reply:
What if we focus some element inside the CssValueInput component that is not the input itself while we drag? This way we still don't have to fight popover focus and we don't have to keep the focus in the input.

Reply:
Doing so we would also need to prevent calling the onBlur event and return the focus to the input once scrubbing is done

Reply:
> Was there no other way? At this point I am not sure which one is worse the jumping caret next to the number or the always selected text.

I thought it looks good actually, didn't try to avoid it. Will try then

Reply:
we had lots of back and forth on that behavior, but take a look at figma, the way they do it is really great, except they don't allow scrub on the input itself, which makes them always need an icon or otherwise there is no scrub.

Reply:
they also support an option key on inputs, which then activates scrub directly on the input, but I bet most people don't know that because its not easy to discover

Reply:
I move caret to the end now, same as Figma

Reply:
We had an extensive prior discussion about this and found it better to not have that carret there. 
Yes figma does that when drag starts on an input field using option key.
But also figma doesn't do that when dragging starts on the icon for fields with the icon.

We made a prior decision to not have that caret when dragging, because it looks cleaner, less annoying.

Have you tried focusing some other element inside the CssValueInput container? Did you exhaust all options to fix it while keeping the behavior we decided on?

Reply:
Focusing other element should work. Didn't try it as I thought moving to the end is fine and it's easier to do. 

Will try focusing other element.

Reply:
works like a charm